service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateVnetStep.java

```diff
@@ -73,20 +72,32 @@ private Network createVnetAndSubnets(
           .withAddressSpace(
               parametersResolver.getValue(
                   CromwellBaseResourcesFactory.ParametersNames.VNET_ADDRESS_SPACE.name()))
-          .withSubnet(
-              CromwellBaseResourcesFactory.Subnet.AKS_SUBNET.name(),
+          .defineSubnet(CromwellBaseResourcesFactory.Subnet.AKS_SUBNET.name())
+          .withAddressPrefix(
               parametersResolver.getValue(CromwellBaseResourcesFactory.Subnet.AKS_SUBNET.name()))
-          .withSubnet(
-              CromwellBaseResourcesFactory.Subnet.BATCH_SUBNET.name(),
+          .withExistingNetworkSecurityGroup(networkSecurityGroupId)
+          .attach()
+          .defineSubnet(CromwellBaseResourcesFactory.Subnet.BATCH_SUBNET.name())
+          .withAddressPrefix(
               parametersResolver.getValue(CromwellBaseResourcesFactory.Subnet.BATCH_SUBNET.name()))
-          .withSubnet(
-              CromwellBaseResourcesFactory.Subnet.POSTGRESQL_SUBNET.name(),
+          .withExistingNetworkSecurityGroup(networkSecurityGroupId)
+          .withAccessFromService(ServiceEndpointType.MICROSOFT_STORAGE)
+          .withAccessFromService(ServiceEndpointType.MICROSOFT_SQL)
+          .attach()
+          .defineSubnet(CromwellBaseResourcesFactory.Subnet.POSTGRESQL_SUBNET.name())
+          .withAddressPrefix(
               parametersResolver.getValue(
                   CromwellBaseResourcesFactory.Subnet.POSTGRESQL_SUBNET.name()))
-          .withSubnet(
-              CromwellBaseResourcesFactory.Subnet.COMPUTE_SUBNET.name(),
+          .withExistingNetworkSecurityGroup(networkSecurityGroupId)
+          .withDelegation("Microsoft.DBforPostgreSQL/flexibleServers")
+          .withAccessFromService(ServiceEndpointType.MICROSOFT_STORAGE)
```

Comment:
Trying to grok what these `withAccessFromService` invocations are for (in concert with this [doc](https://learn.microsoft.com/en-us/azure/virtual-network/virtual-network-service-endpoints-overview)). Is this _required_ for the resources in the `COMPUTE_SUBNET` to be able to access storage resources? Or is it more of an optimization?  Same for the other `withAccessFromService` calls.

Reply:
Its an cost optimization for Batch. The key line from that document is "Service Endpoints enables private IP addresses in the VNet to reach the endpoint of an Azure service without needing a public IP address on the VNet." Usually when Batch compute nodes talk to storage, they pass through a load balancer with a public IP address. The load balancer has an associated network throughput cost, which ends up being quite high.